src/omv/imlib/framebuffer.c

```diff
@@ -327,19 +312,21 @@ void framebuffer_flush_buffers(bool fifo_flush) {
 
 int framebuffer_set_buffers(int32_t n_buffers) {
     uint32_t total_size = framebuffer_raw_buffer_size();
-    uint32_t size = total_size / n_buffers;
+    uint32_t vbuff_size = (n_buffers == 1) ?
+                          total_size : FB_ALIGN_SIZE_ROUND_UP(framebuffer->frame_size + sizeof(vbuffer_t));
+    int32_t max_buffers = (n_buffers == 1) ? n_buffers : (total_size / vbuff_size);
```

Comment:
```suggestion
    uint32_t vbuff_size = (n_buffers == 1) ?
                          FB_ALIGN_SIZE_ROUND_DOWN(total_size -  sizeof(vbuffer_t)) :
                          FB_ALIGN_SIZE_ROUND_UP(framebuffer->frame_size + sizeof(vbuffer_t));
    int32_t max_buffers = total_size / vbuff_size;
```

I think you need to round down the total size as this doesn't include the vbuffer header which would use some space.

Then max_buffers when n_buffers is one should be the same calculation either way.

Reply:
> Then max_buffers when n_buffers is one should be the same calculation either way.

How can `total_size` equal `FB_ALIGN_SIZE_ROUND_DOWN(total_size -  sizeof(vbuffer_t))` for this to work:
```C
     uint32_t vbuff_size = (n_buffers == 1) ?
                          FB_ALIGN_SIZE_ROUND_DOWN(total_size -  sizeof(vbuffer_t)) :
    int32_t max_buffers = total_size / vbuff_size;
```

Reply:
I could rearrange things to make that work though. However, vbuff_size here already includes the header size.

Reply:
vbuff_size would always be slightly less than total_size so the result would be 1. Which was happening before with (n_buffers == 1) ? n_buffers.

Reply:
>vbuff_size would always be slightly less than total_size so the result would be 1

Yes that's right. Well now it's equal so will be 1.